backend/index.ts

```diff
@@ -18,31 +18,162 @@ delete --> delete
 
+//Changes the profile picture of the user once profile has already been created
 app.put('/changeProfilePic', (req, res) => {
   // do this things with that
   const {userID, photoURL} = req.body
   // res -> status code
 });
 
+//Deletes User
+app.delete('/user', (req, res) => {
+  const {userID} = req.body
+  // res -> status code
+});
+
+//////////////////////////////////////////////////////////////////////////////////////COMMUNITY\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
+
+//Create A Community Object
+app.post('/community', (req, res) => {
+  const {userID, communityName, description, picture} = req.body;
+  // res -> status code
+});
+
+//Gets the all the data from a community from the database
+app.get('/community', (req, res) => {
+  const {communityID} = req.query; 
+  // res -> json object
+});
+
+//Allows moderators to edit description of community
+app.put('/editDescription', (req, res) => {
+  const {userID, communityID, description} = req.query; 
+  // res -> status code
+});
+
+//Changes the profile picture of the community
+app.put('/changeCommunityPic', (req, res) => {
+  // do this things with that
+  const {userID, communityID, photoURL} = req.body
+  // res -> status code
+});
+
+//Gives user moderator
+app.put('/makeMod', (req, res) => {
+  // do this things with that
+  const {communityID, userID} = req.body
+  // res -> status code
+});
+
+//Bans User From Community
+app.delete('/banUser', (req, res) => {
+  // do this things with that
+  const {communityID, userID} = req.body
+  // res -> status code
+});
+
+//Deletes Community
+app.delete('/user', (req, res) => {
+  const {userID, communityID} = req.body
+  // res -> status code
+});
+
+//////////////////////////////////////////////////////////////////////////////////////COMMUNITY\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
+
+//Creates A Post
+app.post('/post', (req, res) => {
+  const {userID, title, type, body, tags} = req.body;
+  // res -> status code
+});
+
+//Gets Post Data
+app.get('/post', (req, res) => {
+  const {userID, postID} = req.query;
+  // res -> JSON object
+});
+
+//Edit Body Of Post
+app.put('/post', (req, res) => {
+  const {userID, postID, edit,tags} = req.query;
+  // res -> status code
+});
+
+//Upvotes Post
+app.put('/upvotePost', (req, res) => {
+  const {userID,postID} = req.query;
+  // res -> status code
+});
+
+//Downvotes Post
+app.put('/downvotePost', (req, res) => {
+  const {userID, postID} = req.query;
+  // res -> status code
+});
+
+//Deletes Post
+app.delete('/user', (req, res) => {
+  const {userID, postID} = req.body
+  // res -> status code
+});
+
+//////////////////////////////////////////////////////////////////////////////////////Comments\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
+//Creates A Comment (no reply)
+app.post('/comment', (req, res) => {
+  const {userID, description, postID} = req.body;
```

Comment:
For convention, change description to data. Or if yall think that's not important let me know.